tests/bud.bats

```diff
@@ -1953,6 +1953,28 @@ _EOF
   test -s $root/etc/passwd
 }
 
+@test "build using --layer-label and test labels on intermediate images" {
+  # Remove all images so no intermediate images are present
+  run_buildah rmi --all -f
+  _prefetch alpine
+
+  run_buildah build --no-cache --layers --layer-label somefancylayerlabel=true $WITH_POLICY_JSON -t exp -f $BUDFILES/simple-multi-step/Containerfile
+
+  # Original image must not contain this label
+  run_buildah inspect --format '{{ index .Docker.Config.Labels "somefancylayerlabel"}}' exp
+  expect_output "" "label label on actual image"
+
+  # All intermediate images must have this label
+  run_buildah images -a --format '{{.ID}}'
+  firstintermediateImage=${lines[1]}
+  secondintermediateImage=${lines[2]}
```

Comment:
This looks it could flake.  We're currently relying on an order of the images but I don't think this order is deterministic.

I think it would be better to add a `--filter reference!=exp".

Reply:
Ah nice catch, `--filter intermediate=true` fits well here.